### PR TITLE
Mmd2csw iso - this PR adds few methods and docs to perfom mmd_to_csw-iso translation

### PR DESCRIPTION
**Pull request by epifanio**
The PR adds:
Two python files in the `py_mmd_tools` module:
* `mmd_util.py` - [ setup a logger, check XML and MMD validity ] 
    Note: the logging is always on, potentially can be turned off in a second PR
* `mmd_to_csw_iso.py` - main method to perform MMD -> CSW-ISO conversion, optionally performs validation of the input MMD
   TODO: add validation for the output ISO (need a reference to ISO XSD schema)

A scrip to use the `mmd_to_csw_iso.mmd_to_iso` a method, and a relative readme with usage instructions.
* `script/mmd2iso-csw.py`

Refactoring:
* renamed `scr` to `script`
* adding a `test` directory so to have test in a dedicated folder with the naming convention: `test_modulename`)

### CODE REVIEW

**Review by mortenwh — CHANGES_REQUESTED** [review]
Thanks for the PR. I have a few overall comments:

- Please make sure to fulfill the requirements in the Definition-of-Done (DoD; https://s-enda-documentation.readthedocs.io/en/latest/definition_of_done.html) and general conventions (https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)
- Please remove lines 3-5 in the file headers (author, created and copyright). This should be replaced with a link to the license, as specified in the DoD. Note that we will be many contributors to this code, so it doesn't make sense to list people in the files, since this is error prone. Contributions are evident from the git log.
- Please follow the structure in the main repo; executables in the scr folder, and tests in the file `py_mmd_tools/tests.py` - if you disagree with anything, please add an issue to let people discuss it before we make changes

**Comment by epifanio** [comment]
I opened two iregarding point 3 on the repository structure
- https://github.com/metno/py-mmd-tools/issues/4  
- https://github.com/metno/py-mmd-tools/issues/5

File: py_mmd_tools/mmd_to_csw_iso.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
 """
 Tool for converting metadata from MMD format to ISO format using a specific xslt.
-Author:    Massimo Di Stefano,
-Created:   03.10.2020 (dd.mm.YYYY)
-Copyright: (c) Norwegian Meteorological Institute
-Usage: 
-EXAMPLE: 
+
+ License:
+     This file is part of the S-ENDA-Prototype repository (https://github.com/metno/S-ENDA-Prototype).
```
Comment (mortenwh) [inline py_mmd_tools/mmd_to_csw_iso.py:5]: but this is wrong.. It is part of the py-mmd-tools repository, and should be linked to the license here. The same goes for the other changes.

**Review by mortenwh — CHANGES_REQUESTED** [review]

File: py_mmd_tools/mmd_to_csw_iso.py (hunk 1 above)
Comment (epifanio) [reply py_mmd_tools/mmd_to_csw_iso.py:5]: Ah! - I simply copied the license example from the link to DoD you posted - without reading - I Ill fix it

File: py_mmd_tools/mmd_util.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        logfilepath=config['paths']['logs'].get()
+    except NotFoundError:
+        logfilepath = './logs/'
+    if not pathlib.Path(logfilepath).exists():
+       pathlib.Path(logfilepath).mkdir(parents=True, exist_ok=True)
+    return logfilepath
+
+
+
+import logging
```
Comment (mortenwh) [inline py_mmd_tools/mmd_util.py:30]: already imported - it's a good practice to have all imports at the top of the file to avoid repetition

**Review by mortenwh — CHANGES_REQUESTED** [review]
Tests are still missing, and restructuring of the repo is not part of this task

**Comment by epifanio** [comment]
> Tests are still missing, and restructuring of the repo is not part of this task

For the test, please provide link or add files to want we want to use for:
* example MMD XML file
* ISO version of the example MMD XML file mentioned above
* location where to find/store the above files

regarding refactoring (which affect testing) - I opened (as suggested) a dedicated issue - can you comment on them? 

As it is now the repository is still empty and if you accept the proposed change this PR could be merged without having to rework the file path (which will save time and resources).

File: py_mmd_tools/mmd_util.py (hunk 2 above)
Comment (epifanio) [reply py_mmd_tools/mmd_util.py:30]: All the imports are at the top, that was obviously left there and repeated by-mistake after adding the `get_logpath` method - Did you find any other misplaced import?

**Comment by mortenwh** [comment]
> > Tests are still missing, and restructuring of the repo is not part of this task
> 
> For the test, please provide a link or add files to what we want to use for:
> 
>     * example MMD XML file

You could use the ones from https://github.com/metno/mmd/tree/master/input-examples - I suggest to make a folder `data` in the tests-folder (instead of `tests.py`, and it should be a package) - it's ok for me to make it parallel to the `py_mmd_tools` package, as you have already suggested.

>     * ISO version of the example MMD XML file mentioned above
What do you mean by that?

>     * location where to find/store the above files
If they are related to the tests, they should be stored in a subfolder of `tests/`. 

Note that if we use proper unit testing and mocking, we should not need any data files. That would be the best, but I guess we can give it a bit of slack.

Can you add CI as in MMD as well? Shouldn't be too much work..?

File: py_mmd_tools/mmd_util.py (hunk 2 above)
Comment (mortenwh) [reply py_mmd_tools/mmd_util.py:30]: No, I didn't see any..

**Comment by epifanio** [comment]
> What do you mean by that?
> 
> >

mmd2iso takes an mmd.xml as input and return an iso as output - A test in this context will compare the returned XML against a reference file - You run a test against a reference that you trust. (which is: the ISO version of the example MMD XML file) -- is it clearer?

**Comment by epifanio** [comment]
> You could use the ones from https://github.com/metno/mmd/tree/master/input-examples -

I am using this one https://github.com/metno/mmd/blob/master/input-examples/sentinel-1-mmd.xml

**Comment by epifanio** [comment]
> I suggest <s>to make</s> making a folder `data` in the tests-folder (instead of `tests.py`, and it should be a package)

> If they are related to the tests, they should be stored in a subfolder of tests/.

> Note that if we use proper unit testing and mocking, we should not need any data files. That would be the best, but I guess we can give it a bit of slack.


I already have commits to addressing your previous comments - refactoring (adding directories or moving files) will not be part of this PR -- for this PR I stick with your initial design of having the test file inside the module directory and (as you wanted) I will not add external data. --  I will use a configuration file from where to read the file location needed by the test.

**Comment by epifanio** [comment]
> Can you add CI as in MMD as well? Shouldn't be too much work..?

Not in this PR and not a priority for my working task at the moment.

**Comment by epifanio** [comment]
The latest commit restores the previous repository layout and add a test for the `mmd_to_iso` method. 

* **Notes:**

    * To handle the path to the needed file and to avoid to copy files across repositories I use a configuration library which is called [`confuse`](https://confuse.readthedocs.io/en/latest/)
      The "confuse" library is used to pass the path string for static files consumed in the test.
        
        From the confuse documentation: 
        * On Linux systems, by default, `confuse` reads a configuration file from:
        ```bash
        $HOME/.config/appname/config.yaml # in this code I use  'mdtool' as 'appname'
        ```
        * The `confuse` configuration file uses `yaml` syntax, an example is provided below:
        
        ```yaml
        paths:
            reference_mmd: 'path/to/reference_mmd.xml'
            reference_iso: 'path/to/reference_iso.xml'
            mmd2isocsw: 'path/to/mmd-to-iso.xsl'
            mmd_xsd: 'path/to/mmd.xsd'
            logs: './logs/'
        ```
        Consult the `confuse` documentation for more details. --

        * If you do not want (or can not) use the 'confuse' library:

            1. replace the values of:
        
            ```python
            reference_mmd
            reference_iso
            mmd2iso_xslt
            ```
               with the path strings to the files provided by your testing environment - e.g.: 
            ```python
            reference_mmd='path/to/reference_mmd.xml'
            reference_iso='path/to/reference_iso.xml'
            mmd2isocsw='path/to/mmd-to-iso.xsl'           
            ```
            2. comment/remove the following lines:
        
            ```python
            import confuse
            config = confuse.Configuration('mmdtool', __name__)
            ```

    * example file used in this test are from the [mmd_repository](), for convenience can be also found grouped as [gist](https://gist.github.com/934db4d3cf4e7a52985a3e231e0d36cf) https://gist.github.com/934db4d3cf4e7a52985a3e231e0d36cf)
    
    * to run the test you need to cd into the module and run the following command.
    ````bash
    python3 -m unittest test_mmd_to_iso.py
    ````
    make sure your environment has access to a configuration file (or replace the path as described above) and that the module is importable

**Review by mortenwh — APPROVED** [review]

**Pull request merged**